### PR TITLE
[Blog post] Our journey from VPC peering to Transient gateway

### PR DESCRIPTION
**Pull request by Puneeth-n**

### CODE REVIEW

File: _posts/2019-07-15-vpc_peering_to_transient_vpcs.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+<center>
+<figure>
+  <img style="width: 100.0%; height: 100.0%" src="/images/2019_07_15/current_setup.png">
+  <figcaption><b>Current VPC architecture at Comtravo</b> <br></figcaption>
+</figure>
+</center>
+
+
+In the next blog post, we look into how to leverage transit gateways and client ot site VPN endpoints to have a centralized access management to the cloud.
```
Comment (Puneeth-n) [inline _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]:
```suggestion
In the next blog post, we look into how to leverage transit gateways and client to site VPN endpoints to have a centralized access management to the cloud.
```

File: _posts/2019-07-15-vpc_peering_to_transient_vpcs.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+comments: false
+author: puneeth_n
+image:
+  teaser: 2019_07_15/aws_logo.png
+  feature: 2019_07_15/tgw.png
+---
+
+We at Comtravo run our services in the cloud with [AWS](https://aws.amazon.com/). All our environments (test and production) are mostly orchestrated by [Terraform](https://www.terraform.io/). Our environments spread across multiple AWS accounts all part of the same organization.
+
+We maintain [VPC](https://aws.amazon.com/vpc/) level isolation between environments. In our acceptance testing and production environments, there exists VPC level isolation between the databases and stateless microservices. In these environments, we use [VPC-peering](https://docs.aws.amazon.com/vpc/latest/peering/what-is-vpc-peering.html) to establish connections between our stateless VPC and stateful VPC so that our stateless services can connect to our databases. The VPCs in these environments which contain databases are not terraformed. Furthermore, all our microservices are dockerized and run on [AWS ECS](https://aws.amazon.com/ecs/).
```
Comment (mattilyra) [inline _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: So, the environments are separated into VPCs and inside the envs the DBs and stateless services are further separated into VPCs?

Comment (mattilyra) [inline _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: Are the last two notes about the DBs not being terraformed and docker/ECS relevant for the rest of the article

[the remark this answers is not in the document]
Comment (Puneeth-n) [reply _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: used

[the remark this answers is not in the document]
Comment (Puneeth-n) [reply _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: `qa`

File: _posts/2019-07-15-vpc_peering_to_transient_vpcs.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+### VPC peerings and Transit VPCs
+
+Transit VPCs are yet another approach which helps in managing lots of independent VPCs and in reducing the attack surface. There are mainly two broad architectures with transit VPCs and which one to use depends upon the use case.
+
+#### Hub and Spoke architecture
+
+If managing the test environments is the only concern, a Hub and Spoke model makes most sense. In this approach, a single VPC called the transit VPC establishes point to point connection to all the VPCS. A user logs in to this transit VPC and could ***potentially*** have access to all the environments. This transit VPC could run a bastion host or a vpn server to which users connect it. This architecture immensely reduces the attack radius and doesn't increase when number of test environments increase. In this architecture the transit VPC is the hub and each VPC peering that it establishes becomes the spoke of the architecture.
```
[the remark this answers is not in the document]
Comment (Puneeth-n) [reply _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: By attack radius I mean points of entry into our cloud infra

File: _posts/2019-07-15-vpc_peering_to_transient_vpcs.md (hunk 2 above)
Comment (Puneeth-n) [reply _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: in acceptance testing (qa) and production environment, databases and stateless services are in different VPCs and were connected via vpc peering but now connected via transit gateways.

Comment (Puneeth-n) [reply _posts/2019-07-15-vpc_peering_to_transient_vpcs.md]: yes

**Comment by Puneeth-n** [comment]
@Juliane5 Thanks for the review ;) ok to merge?

**Pull request merged**